src/lib/Libifl/pbsD_deljoblist.c

```diff
@@ -46,160 +46,554 @@
+ *
+ * @param[in]	fd - connection handle
+ *
+ * @return	struct batch_deljob_status *
+ * @retval	List of error objects from server (or NULL)
+ */
+static struct batch_deljob_status *
+recv_deljob(int fd)
+{
+	struct batch_reply *reply;
+	struct batch_deljob_status *ret = NULL;
 
-	if (reply == NULL && pbs_errno == PBSE_NONE) {
+	reply = PBSD_rdrpy(fd);
+	if (reply == NULL && pbs_errno == PBSE_NONE)
 		pbs_errno = PBSE_PROTOCOL;
-	} else if (reply->brp_choice != BATCH_REPLY_CHOICE_NULL  &&
-		reply->brp_choice != BATCH_REPLY_CHOICE_Text &&
-		reply->brp_choice != BATCH_REPLY_CHOICE_Delete) {
+	else if (reply->brp_choice != BATCH_REPLY_CHOICE_NULL &&
+		 reply->brp_choice != BATCH_REPLY_CHOICE_Text &&
+		 reply->brp_choice != BATCH_REPLY_CHOICE_Delete)
 		pbs_errno = PBSE_PROTOCOL;
-	} 
-	
+
 	if ((reply != NULL) && (reply->brp_un.brp_deletejoblist.brp_delstatc != NULL)) {
-		rbsp = reply->brp_un.brp_deletejoblist.brp_delstatc;
+		ret = reply->brp_un.brp_deletejoblist.brp_delstatc;
 		reply->brp_un.brp_deletejoblist.brp_delstatc = NULL;
 	}
-	
 	PBSD_FreeReply(reply);
 
-	/* unlock the thread lock and update the thread context data */
-	if (pbs_client_thread_unlock_connection(c) != 0)
+	return ret;
+}
+
+/**
+ * @brief	Convenience function to send deljoblist request and receive reply
+ *
+ * @param[in] fd - communication handle
+ * @param[in] function - req type
+ * @param[in] jobids - the job id list
+ * @param[in] numjids - the count of job ids
+ * @param[in] extend - extend string for req
+ *
+ * @return struct batch_deljob_status *
+ * @retval list of error objects from server (or NULL)
+ */
+static struct batch_deljob_status *
+deljob_to_server(int fd, int function, char **jobids, int numids, char *extend)
+{
+	if (send_deljob(fd, function, jobids, numids, extend))
 		return NULL;
 
-	return rbsp;
+	return recv_deljob(fd);
 }
 
 /**
- * @brief
- *      -encode a Delete Job List Batch Request
+ * @brief	Broadcast deljob request to other servers in a multi-server set up
+ * 			This handles the cases where a job was moved to another server
+ *
+ * @param[in] c - connection handle (virtual)
+ * @param[in] function - req type
+ * @param[in] jobids - the job id list
+ * @param[in] numjids - the count of job ids
+ * @param[in] extend - extend string for req
  *
- * @par Functionality:
- *              This request is used for delete operation.
- *              
+ * @return struct batch_deljob_status *
+ * @retval list of jobs not found on any server
+ */
+static struct batch_deljob_status *
+broadcast_deljob(int c, int function, char **jobids, int numids, char *extend)
+{
+	int *skip_list;
+	svr_conn_t **conns = get_conn_svr_instances(c);
+	int num_svrs = get_num_servers();
+	int i;
+	struct batch_deljob_status **svr_retlist = NULL;
+	struct batch_deljob_status *retlist = NULL;
+	struct batch_deljob_status *iter_list = NULL;
+	struct batch_deljob_status *prev = NULL;
+	struct batch_deljob_status *next = NULL;
+
+	if (conns == NULL) {
```

Comment:
get_conn_svr_instances will return NULL for instance fd.

Reply:
We will not broadcast for an instance fd case. So, once inside this function, conns should not be NULL.